src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -1030,7 +1037,7 @@ REDHAWK_PALEXPORT void REDHAWK_PALAPI PalHijack(HANDLE hThread, _In_opt_ void* p
     }
 #endif
 
-    if ((status != 0) && (status != EAGAIN))
+    if ((status != 0) && (status != EAGAIN) && (status != ESRCH))
```

Comment:
Why are we getting ESRCH here?

Reply:
This was happening in the threading smoke test. I assume it happens if the thread is dying at the same time as we try to suspend it (but I did not retry with your GC fix). Mono handled the error code so I assumed that it makes sense to do it here too.

Reply:
I can still hit it even with #75298 applied.

Reply:
It is suspisious. We are holding threadstore lock at this point, so a thread can't exit.
 
Perhaps GC crash causes rude termination of a thread, which leaves threadstore corrupted. 
I such case I am not sure if ignoring is better than aborting as we could get stuck trying to suspend a thread that does not exist. 
Let's see if this happens after the fix.

Reply:
I can check it under debugger now that I know what to look for.

Reply:
> I can still hit it even with https://github.com/dotnet/runtime/pull/75298 applied.

Ok, you can keep it as a workaround, but this needs to be investigated. Maybe there is another more transient reason, besides "the thread is dead" for this.

Reply:
The thread that is being signalled has this stack trace when it happens:
```
  thread #66
    frame #0: 0x00000001003dfe2c UnitTests`_S_P_CoreLib_System_Threading_WaitSubsystem_HandleManager__FromHandle(handle=4330599712) at WaitSubsystem.HandleManager.Unix.cs:29
    frame #1: 0x00000001002b7cdc UnitTests`_S_P_CoreLib_System_Threading_WaitSubsystem__SetEvent(handle=4330599712) at WaitSubsystem.Unix.cs:244
    frame #2: 0x00000001002a8328 UnitTests`_S_P_CoreLib_System_Threading_EventWaitHandle__Set(this=0x00000001047b1838) at EventWaitHandle.Unix.cs:48
    frame #3: 0x00000001002a4100 UnitTests`_S_P_CoreLib_System_Threading_Thread__OnThreadExit at Thread.NativeAot.Unix.cs:53
    frame #4: 0x00000001000a5220 UnitTests`::PalDetachThread(thread=0x00000001025044b0) at PalRedhawkUnix.cpp:502:9
    frame #5: 0x000000010002cdec UnitTests`ThreadStore::DetachCurrentThread(shutdownStarted=false) at threadstore.cpp:162:10
    frame #6: 0x0000000100027e6c UnitTests`RuntimeThreadShutdown(thread=0x00000001025044b0) at startup.cpp:452:5
    frame #7: 0x00000001000a6edc UnitTests`TlsDestructionMonitor::~TlsDestructionMonitor(this=0x00000001025045a8) at PalRedhawkUnix.cpp:451:13
    frame #8: 0x00000001000a5158 UnitTests`TlsDestructionMonitor::~TlsDestructionMonitor(this=0x00000001025045a8) at PalRedhawkUnix.cpp:448:5
    frame #9: 0x00000001a0085cc0 dyld`invocation function for block in dyld4::RuntimeState::_finalizeListTLV(void*) + 56
    frame #10: 0x00000001a0085c28 dyld`dyld4::RuntimeState::_finalizeListTLV(void*) + 116
    frame #11: 0x00000001a039a968 libsystem_pthread.dylib`_pthread_tsd_cleanup + 488
    frame #12: 0x00000001a039d760 libsystem_pthread.dylib`_pthread_exit + 84
    frame #13: 0x00000001a039d078 libsystem_pthread.dylib`_pthread_start + 160
```

Reply:
FWIW this is on `macOS Ventura 13.0 Beta (22A5321d)`. It could be some pthread behavior specific to it.

Reply:
The thread is exiting gracefully and in the process of being trapped on the thread store lock. I guess OS already considers it dead and does not want to or cannot deliver signals to it.   

We would need to allow `ESRCH` on OSX - the same way we allow `ENOTSUP`.

The thread is not running managed code and we will ignore it when we retry hijacking, so it is ok to ignore.